flink-table/flink-table-common/src/main/java/org/apache/flink/table/functions/BuiltInFunctionDefinitions.java

```diff
@@ -275,6 +275,20 @@ ANY, and(logical(LogicalTypeRoot.BOOLEAN), LITERAL)
                             "org.apache.flink.table.runtime.functions.scalar.ArrayReverseFunction")
                     .build();
 
+    public static final BuiltInFunctionDefinition ARRAY_SLICE =
+            BuiltInFunctionDefinition.newBuilder()
+                    .name("ARRAY_SLICE")
+                    .kind(SCALAR)
+                    .inputTypeStrategy(
+                            sequence(
+                                    logical(LogicalTypeRoot.ARRAY),
+                                    logical(LogicalTypeRoot.INTEGER),
+                                    logical(LogicalTypeRoot.INTEGER)))
+                    .outputTypeStrategy(nullableIfArgs(argument(0)))
```

Comment:
As far far i remember in jira it was mentioned that the third arg could be optional...

Reply:
no, it is not an optional.

Reply:
https://issues.apache.org/jira/browse/FLINK-32260

Reply:
can we have it optional as it was suggested in the mentioned jira's issue
or is there any blocker for that?

Reply:
I saw the jira again, one one suggest third arg could be optional.

Reply:
> can we have it optional as it was suggested in the mentioned jira's issue or is there any blocker for that?

The current description of this function is already very complicated. If the third args becomes optional again, is it too complicated for users?